packages/hmr-plugin/src/internal/hmr-lifecycle.ts

```diff
@@ -69,4 +71,14 @@ export class HmrLifecycle<T extends Partial<NgxsHmrLifeCycle<S>>, S> {
       });
     });
   }
+
+  private detectIvyWithJIT(): void {
+    const jit: boolean = this.ngAppModule.constructor.hasOwnProperty('__annotations__');
+    const isIvyMode: boolean =
+      this.options.isIvyMode === undefined ? isIvy() : this.options.isIvyMode;
+
+    if (isIvyMode && jit) {
+      console.error(`@ngxs/hrm-plugin: not working with JIT mode in Angular Ivy`);
```

Comment:
```suggestion
      console.error(`@ngxs/hrm-plugin doesn't work with JIT mode in Angular Ivy. Please use AOT mode.`);
```

Reply:
PS. do you think that we should use `console.error` here, or perhaps throw an error so that the issue isn't hidden in the console?

Reply:
> PS. do you think that we should use `console.error` here, or perhaps throw an error so that the issue isn't hidden in the console?

What is your opinion on this?

Reply:
@markwhitfeld Actually, I would rather output an exception. But in any case, this does not affect the overall execution flow, since this code is executed after ngxsAfterBootstrap.